tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,154 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define DIRECTORY "/fluent-bit/buffer/s3"
```

Comment:
how about BUFFER_DIRECTORY, to be more clear.

Ideally should also be in the local tests directory- see example here: https://github.com/fluent/fluent-bit/blob/master/tests/internal/aws_credentials_profile.c#L10

Reply:
Actually, it needs to be that, otherwise test process might not have permissions to write to the buffer directory, or it might not exist. Specifically, do:

```
#define BUFFER_DIRECTORY FLB_TESTS_DATA_PATH "data/s3_local_buffer/"
```

And then create that directory and add gitkeep file to it, as I have done here: https://github.com/PettitWesley/fluent-bit/commit/93e389ff899fcb41b0c0cfef828aad0fb8802c0e